File: Core/Entities.cs
namespace Core;

public class Post
{
    public int Id { get; set; }
    public int Category { get; set; }
    public double Price { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public DateTime Date { get; set; }
    //Connection to User Table
    public int UserId { get; set; }
    public User User { get; set; } 
    public string Img { get; set; }
    public virtual List<Comment>? Comments { get; set; }
}

public class User
{
    public int Id { get; set; }
    public string userName { get; set; }
    public string Email { get; set; }
    public string PasswordHash { get; set; }
    public string Salt { get; set; }
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public string Address { get; set; }
    public string PostalCode { get; set; }
    public string PhoneNumber { get; set; }
    public virtual List<Post>? Posts { get; set; }
    // public int RoleId { get; set; }
    public string? Img { get; set; }
    public int RoleId { get; set; }
    
    //public byte[] Image { get; set; }
    //public List<Roles>? Roles { get; set; }
}

public class Comment
{
    public int Id { get; set; }
    public string Content { get; set; }
    
    public int PostID { get; set; }
    public Post Post { get; set; }
}

File: frontend/src/entities/entities.ts
export interface registerDTO{
  email:string,
  password:string,
  firstName:string,
  lastName:string,
  userName:string,
  address:string,
  postalCode:string,
  phoneNumber:string,
  img?:string,
  roleId:number
}
export interface CommentDTO {
  postId:number,
  content:string,
  author:string,
  date:string,
  avatar:string
}
export interface loginDTO{
  Email: string,
  Password: string
}
export interface postDTO{
  id:number,
  email:string,
  userName:string,
  price:number,
  title:string,
  description:string,
  authority:number,
  address:string,
  category:number
  img:string
}
export interface MailDTO {
  receiver:string,
  sender:string,
  subject:string,
  mail_content:string,
  receiverName:string,
  senderName:string
}
export interface createPostDTO {
  Email:string,
  Title: string,
  Description: string,
  Price: number,
  Category: number,
  Img:string
}
export interface sessionToken{
  exp?: number;
  role?: number;
  userName?:string;
  email?:string;
}

export interface Category{
  id:number;
  name:string;
  ticked: boolean;
}

export interface NavBarSearch {
  name: string;
  ticked: boolean;
  categories: Category[];
}

export interface categoryDTO{
  ids: number[]
}

export interface priceDTO{
  min: number,
  max: number;
}

export interface catPriceDTO{
  ids: number[],
  min: number,
  max: number
}

export interface searchDTO{
  args: string
}

export interface filterSearchDTO{
  operationType: number,
  dto: categoryDTO | priceDTO | catPriceDTO | searchDTO
}
export interface UserProperties{
  userName:string,
  email:string,
  roleId:number
}

